src/entities/friendships.rs

```diff
@@ -206,6 +221,49 @@ impl FriendshipRepositoryImplementation for FriendshipsRepository {
         }
     }
 
+    #[tracing::instrument(name = "Get mutual user friends from DB")]
+    async fn get_mutual_friends<'a>(
+        &'a self,
+        address_1: &'a str,
+        address_2: &'a str,
+        transaction: Option<Transaction<'a, Postgres>>,
+    ) -> (
+        Result<Vec<String>, sqlx::Error>,
+        Option<Transaction<'a, Postgres>>,
+    ) {
+        let query = MUTUALS_FRIENDS_QUERY.to_string();
+
+        let query = sqlx::query(&query).bind(address_1).bind(address_2);
+
+        let executor = self.get_executor(transaction);
+
+        let (res, resulting_executor) = DatabaseComponent::fetch_all(query, executor).await;
+
+        let transaction_to_return = get_transaction_result_from_executor(resulting_executor);
+
+        match res {
+            Ok(rows) => {
+                let response = Ok(rows
+                    .iter()
+                    .map(|row| -> String { row.try_get("address").unwrap() })
```

Comment:
Do we need the explicit `-> String` when a line below we are already making the types explicit for the return?

Reply:
No, removed it